pkgs/build-support/rust/default.nix

```diff
@@ -93,8 +93,11 @@ in stdenv.mkDerivation (args // {
 
   installPhase = args.installPhase or ''
     runHook preInstall
-    mkdir -p $out/bin
-    find target/release -maxdepth 1 -executable -type f -exec cp "{}" $out/bin \;
+    mkdir -p $out/bin $out/lib
+    find target/release -maxdepth 1 -type f -executable ! \( -regex ".*.\(so.[0-9]\|so\|a\|dylib\)" \) -print0 | xargs -r -0 cp -t $out/bin
+    find target/release -maxdepth 1 -regex ".*.\(so.[0-9]\|so\|a\|dylib\)" -print0 | xargs -r -0 cp -t $out/lib
```

Comment:
Not sure what the rust does but openssl has more complex sonames: libcrypto.so.1.0.0 

Should be covered by

```
so.[0-9.]\+
```

I hope.

Reply:
Does find match the whole string with the regex or does it need a `$` at the end?

Reply:
I changed it to `[0-9.]+` since the escape char breaks it. According to `man` find matches on the whole path.